### PR TITLE
Add change-password admin command

### PR DESCRIPTION
**Pull request by strk**
Not working yet, need help

### CODE REVIEW

File: cmd/admin.go
Comment (ptman) [inline cmd/admin.go]: It's not being printed.

**Comment by strk** [comment]
It is now:

$ ./gitea admin --help | grep change
     change-password  Print encoded version of a given password

File: cmd/admin.go
Comment (ptman) [reply cmd/admin.go]: I meant, that the encoded password isn't being printed. Should the usage not just be "change password of a user" ?

**Review by appleboy — APPROVED** [review]

**Comment by appleboy** [comment]
LGTM

File: cmd/admin.go
Comment (strk) [reply cmd/admin.go]: Ah, of course, will fix and squash

**Comment by strk** [comment]
@ptman better now ?

Do you think an error should be printed for users whose LoginSource != LOCAL/PLAIN ?
(but note that the OAuth2 users still have a broken LoginSource: #1143)

**Comment by ptman** [comment]
Yes, much better. And you're right about giving an error for users where it wouldn't be effective or some other method may still allow them to log in (if trying to disable with password change)

**Comment by strk** [comment]
@ptman I'd postpone that part to after  #1143 is done, or I'd have to add more conditionals there to deal with OAuth2 in a special way, which I'd hate...

File: cmd/admin.go
Comment (lunny) [inline cmd/admin.go:63]: Why not just `password`?

Comment (strk) [reply cmd/admin.go:63]: No reason, this was suggested by @bkcsoft on IRC if I recall correctly, and I thought it was a clean and self-describing name

**Comment by pgaskin** [comment]
LGTM

File: cmd/admin.go
Comment (lunny) [reply cmd/admin.go:63]: `change-password` is too long for a command, I think `password` is better!

**Comment by strk** [comment]
You also have to type --username and --password so it is balanced to me
--strk;

Sent from hand-held device, sorry for brevity

File: cmd/admin.go
Comment (appleboy) [reply cmd/admin.go:63]: vote `password`

File: cmd/admin.go
Comment (appleboy) [inline cmd/admin.go]: support `-u` or `--username`. Maybe `username,u`

File: cmd/admin.go
Comment (appleboy) [inline cmd/admin.go]: support `-p` or `--password`. Maybe `password,p`

**Comment by lunny** [comment]
agree with @appleboy

**Comment by strk** [comment]
This PR has 2 LGTM, please merge

**Comment by appleboy** [comment]
@strk  Do you see my suggestions about adding more flags?

**Comment by strk** [comment]
@appleboy added -u and -p options, and rebased

**Comment by strk** [comment]
so I'm also happy with just "passwd" as the admin name (to reflect the unix command) but as the only existing command is `create-user` I think it's consistent to have `change-password` as is. Can it be merged now ? (the rules say 2 LGTM are enough to merge...)

File: cmd/admin.go
Comment (tboerger) [inline cmd/admin.go]: Pretty useless usage information.

File: cmd/admin.go
Comment (tboerger) [inline cmd/admin.go]: Pretty useless usage information.

**Review by tboerger — COMMENTED** [review]
Generally I would prefer a user sub-command instead of admin.

**Comment by strk** [comment]
If every single maintainer complains about something this will become much slower than Gogs. It's already happening that Gogs is ahead. This PR got 2 positive reviews, so about the rules it should just be merged.

If you think maintainers should also have veto powers so please add a section about that in the CONTRIBUTING.md file.

File: cmd/admin.go
Comment (bkcsoft) [reply cmd/admin.go:63]: @lunny @appleboy we already have `create-user` so I don't really see the issue with `change-password`. Then it becomes self-explaining :)

**Comment by bkcsoft** [comment]
@tboerger don't really see the point of having `gitea user [...]` since they are inheritly admin commands 😉

**Comment by bkcsoft** [comment]
LGTM working?

**Comment by appleboy** [comment]
LGTM is working but waiting build status.

**Comment by appleboy** [comment]
@bkcsoft Thanks for improving usage information

**Pull request merged**
